### PR TITLE
[CURATOR-565] check useContainerParentsIfAvailable for ExistsBuilderImpl.

### PR DESCRIPTION
**Pull request by PhantomThief**

### CODE REVIEW

**Comment by Randgalt** [comment]
Please provide a test case and/or more information on why this is needed.

File: curator-framework/src/main/java/org/apache/curator/framework/imps/ExistsBuilderImpl.java [hunk 1]
```diff
@@ -67,7 +67,9 @@ public ACLableExistBuilderMain creatingParentsIfNeeded()
     @Override
     public ACLableExistBuilderMain creatingParentContainersIfNeeded()
     {
-        createParentContainersIfNeeded = true;
+        if (client.useContainerParentsIfAvailable()) {
```
Comment (Randgalt) [inline curator-framework/src/main/java/org/apache/curator/framework/imps/ExistsBuilderImpl.java:70]: this should be on the `creatingParentsIfNeeded()` method

**Comment by eolivelli** [comment]
@PhantomThief do you have cycles to follow up on this patch ?

**Comment by tisonkun** [comment]
@PhantomThief this patch may make sense as we use different logic handle these flags between `CreateBuilder` and `ExistBuilder`. However, it may not be resolved in this way but align to how `CreateBuilderImpl` handles `createParentsIfNeeded` and `createParentsAsContainers`.

**Comment by tisonkun** [comment]
Closed as stale. Feel free to resubmit it if you're still willing to proceed.

**Pull request closed without merging**